manim/mobject/types/vectorized_mobject.py

```diff
@@ -883,26 +921,27 @@ def change_anchor_mode(self, mode: str):
         """
         assert mode in ["jagged", "smooth"]
         nppcc = self.n_points_per_cubic_curve
+
         for submob in self.family_members_with_points():
-            subpaths = submob.get_subpaths()
-            submob.clear_points()
-            # A subpath can be composed of several bezier curves.
-            for subpath in subpaths:
-                # This will retrieve the anchors of the subpath, by selecting every n element in the array subpath
-                # The append is needed as the last element is not reached when slicing with numpy.
-                anchors = np.append(subpath[::nppcc], subpath[-1:], 0)
-                if mode == "smooth":
+            # Every submobject will have its handles modified
+            if mode == "jagged":
+                # The following will make the handles aligned with the anchors,
+                # thus making the Bézier curves straight lines
+                starts = submob.get_start_anchors()
+                ends = submob.get_end_anchors()
+                for a in self.bezier_alphas:
+                    submob.points[1::nppcc] = interpolate(starts, ends, a)
```

Comment:
Yes, my bad - going to correct that!

Reply:
Fixed! There was also some bad slicing going on, which I didn't notice before: `submob.points[1::nppcc]`, the first group of handles, was getting incorrectly reassigned every time, instead of assigning the handle groups at the correct positions! It should have been something like `submob.points[i::nppcc]`, with an `i` instead of an `1`. I ended up using a `for i in range(1, nppcc - 1):` to handle this.